### DISCUSSION TITLE
Is glaze good for network message serialization?

### DISCUSSION THREAD

**Discussion by clocky-dev** [stephenberry/glaze#735] [Q&A] [answered]
Hello,
I wanted to ask if the glaze is good enough for using networking to submit message fast. 

I saw glaze supports BVE, should I use that for message serialization? Is it faster than JSON (glaze)? 

Thanks for your help.

Best regards

**Comment by stephenberry** [comment, marked as answer]
Yes, Glaze is used by many for networking (including the company I work for). Sending JSON messages keeps things human readable and easy for other programs, languages, and clients to interface with. BEVE is a good deal faster than JSON (2X to 3X) and the messages will be smaller. But, BEVE is new and does not have wide support like JSON, so I would only recommend it if you are handling both the server and the client.

FYI, Glaze will be releasing a new RPC interface for JSON and BEVE soon. But, Glaze also supports JSON RPC 2.0 (see documentation).

**Reply by clocky-dev** [reply]
I see but in general, BEVE supports also the same meta data as glaze? So I can just change read_json & to read_binary and write_json to read_binary? 

Thanks!

**Reply by stephenberry** [reply]
Yes, you should be able to simple swap `read_json` to `read_binary`, and the same for writing. The goal is for the two to behave the same, except for format. Although, JSON currently has a bit more features supported than BEVE.

**Comment by clocky-dev** [comment]
After switching from read_json to read_binary this "case" dont work anymore, any idea to tell this is a null object or smth like that? 
```c++
		struct DownloadLogs
		{
			NETWORK_META_DATA(DownloadLogs)

			struct glaze {
				using T = DownloadLogs;
				static constexpr auto value = glz::object();
			};
		}; 

                DownloadLogs value ;
                glz::read_binary(value, ...); // fails here with string buffer
```

**Reply by stephenberry** [reply]
Thanks for pointing out this bug. I just merged a fix (#736), so it should work now for you.

**Reply by clocky-dev** [reply]
You're the best thanks!
